### PR TITLE
Support Glide defaults, and aliases in presets

### PR DESCRIPTION
**Pull request by jackmcdade**
Closes statamic/ideas#616
Closes statamic/ideas#898

### CODE REVIEW

File: config/assets.php [hunk 1]
```diff
@@ -55,6 +55,21 @@
...
+        |--------------------------------------------------------------------------
+        |
+        | You may define global defaults for all manipulation parameters, such as
+        | quality, format, and sharpness. These can and will be be overwritten
+        | on the tag parameter level as well as the preset level.
+        |
+        */
+
+        'defaults' => [
+            'quality' => 1,
```
Comment (jasonvarga) [inline config/assets.php]: You want the default to be 1? Maybe we should just leave it commented out.

```suggestion
            // 'quality' => 1,
```

File: src/Imaging/ImageGenerator.php [hunk 2]
```diff
@@ -278,18 +278,53 @@ private function generate($image)
...
+            'filter' => 'filt',
+            'format' => 'fm',
+            'gamma' => 'gam',
+            'height' => 'h',
+            'orientation' => 'or',
+            'pixelate' => 'pixel',
+            'quality' => 'q',
+            'sharpen' => 'sharp',
+            'width' => 'w',
+            'watermark' => 'mark',
```
Comment (jasonvarga) [inline src/Imaging/ImageGenerator.php:315]: This is neat, but you can't use our tag aliases within presets like you mention in the comments. Should we just make people use the standard glide api params like the presets to be consistent?

**Review by jasonvarga — CHANGES_REQUESTED** [review]

File: config/assets.php (hunk 1 above)
Comment (jasonvarga) [reply config/assets.php]: Oops you already fixed this.

File: src/Imaging/ImageGenerator.php (hunk 2 above)
Comment (jasonvarga) [reply src/Imaging/ImageGenerator.php:315]: Actually we could probably add support for the aliases in the presets by using the same mapping you've just done.

Comment (jackmcdade) [reply src/Imaging/ImageGenerator.php:315]: What about running the presets through the same method?

Comment (jesseleite) [reply src/Imaging/ImageGenerator.php:315]: Make it `public static` if other classes need it, since it's not a stateful helper 🔥

Comment (jasonvarga) [reply src/Imaging/ImageGenerator.php:315]: Yeah I'd add a method to `GlideManager.php` - something like `normalizeParameters()` and move the mapping in there, so you could do `Statamic\Facades\Glide::normalizeParameters($defaults)` and `Glide::normalizeParameters($presets)` [on this line](https://github.com/statamic/cms/blob/15d1b80b18860b0b92f5756f1e917838ac1a8876/src/Imaging/Manager.php#L57).

**Pull request merged**